pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -1215,3 +1207,50 @@ def track_progress(xk):
 
         else:
             raise MinimizeException(f"Optimization failed: {result.message}")
+
+
+class CustomModelWrapper(BaseModel):
+    """Wrapper for the BudgetOptimizer to handle custom PyMC models."""
+
+    model_config = ConfigDict(arbitrary_types_allowed=True, extra="forbid")
+
+    base_model: Model = Field(
+        ...,
+        description="Underlying PyMC model to be cloned for optimization.",
+    )
+    idata: InferenceData
+    channel_columns: list[str] = Field(
+        ...,
+        description="Channel labels used for budget optimization.",
+    )
+    adstock: Any = Field(
+        default_factory=lambda: type("Adstock", (), {"l_max": 0})(),
```

Comment:
Where is this used?
Can you pull this out or create a protocol for this object. Does it make sense for there to be a default?